packages/api/src/routers/conferencing.ts

```diff
@@ -0,0 +1,152 @@
+import { TRPCError } from "@trpc/server";
+import { Temporal } from "temporal-polyfill";
+import { z } from "zod";
+
+import { accountToConferencingProvider, accountToProvider } from "../providers";
+import { createTRPCRouter, protectedProcedure } from "../trpc";
+import { getAccounts, getDefaultAccount } from "../utils/accounts";
+
+export const conferencingRouter = createTRPCRouter({
+  create: protectedProcedure
+    .input(
+      z.object({
+        calendarId: z.string().optional(),
+        eventId: z.string().optional(),
+        accountId: z.string(),
+        calendarAccountId: z.string().optional(),
+        providerId: z.enum(["google", "zoom", "none"]).default("none"),
+        agenda: z.string().default("Meeting"),
+        startTime: z.string(),
+        endTime: z.string(),
+        timeZone: z.string().optional(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      if (input.providerId === "none") {
+        if (input.calendarId && input.eventId && input.accountId) {
+          const accounts = await getAccounts(ctx.user, ctx.headers);
+          const account = accounts.find((a) => a.id === input.accountId);
+
+          if (account) {
+            try {
+              const calendarProvider = accountToProvider(account);
+              const calendars = await calendarProvider.calendars();
+              const calendar = calendars.find((c) => c.id === input.calendarId);
+
+              if (calendar) {
```

Comment:
try try using short circuiting (i.e. negate ! the condition when possible) with early returns

this way you avoid deeply nested statements